src/golang/lib/engine/aq_engine.go

```diff
@@ -717,10 +724,65 @@ func (eng *aqEngine) cleanupWorkflow(ctx context.Context, workflowDag dag_utils.
 	}
 }
 
+func (eng *aqEngine) executeWithEngine(
```

Comment:
can you link to a todo linear task about how we'll want to change this so that we have a separate databricks `orchestrator`

Reply:
https://linear.app/aqueducthq/issue/ENG-2295/refactor-executeworkflow-to-abstract-out-the-orchestration-step